otx/cli/tools/build.py

```diff
@@ -49,29 +53,88 @@ def main():
     """Main function for model or backbone or task building."""
 
     args = parse_args()
+    args.task = args.task.upper() if args.task is not None else args.task
 
     builder = Builder()
 
-    # Build with task_type -> Create User workspace
     otx_root = get_otx_root_path()
+    if args.workspace_root is None:
+        args.workspace_root = f"./otx-workspace-{args.task}"
+        if args.model:
+            args.workspace_root += f"-{args.model}"
+    workspace_path = Path(args.workspace_root)
+
+    # Auto configuration & Datumaro Helper
+    # First, find the format of train dataset and import it by using DatasetManager
+    # Second, find the task type
+    # Third, check the whether val_data_roots is None or not None
+    # If val_data_roots is None, auto_split() is trigerred.
+    # Forth, save the dataset with it's configuraiton(.yaml) file
+    is_autoconfig_enabled = args.train_data_roots
```

Comment:
I think we need to define a different condition to enable auto-config. When we don't have template passed to the CLI and train-data-root is available. If we use only train-data-root checking it is not right, since we can have template path and val-data-roots as well to run ordinary training

Reply:
This block is dependent on `template` and `task` arguments because it just find the `data_format`, `task_type`, and `auto_split` and it'll ignore auto-detected `task_format` if user input their `--task` argument directly.

Let's assume the user's input arguments.  
Left: User's input argument, Right: configuration setting (A: auto-config, U: user input)

1. --train-data-roots --> train dataset (A) + val dataset (A) + task_type (A) + template (A)
2. --train-data-roots + --val-data-roots --> train dataset (U) + val dataset (U) + task_type (A) + template (A)
3. --train-data-roots + --task --> train_dataset (A) + val_dataset (A) + task_type (U) + template (A)
4. --train-data--roots + --task + --template --> train_dataset (A) + val_dataset (A) + task_type (U) + template (U)

Reply:
I still didn't understand that `train_data_roots` is always set to autoconfig when it comes in.
If `--task` and `--train-data-roots` come in together, wouldn't it be possible to reduce task detection?
Isn't the time complexity of the function that detects the task in the dataset high?

Reply:
> sorry, but I still didn't understand that `train_data_roots` is always set to autoconfig when it comes in. If `--task` and `--train-data-roots` come in together, wouldn't it be possible to reduce task detection? Isn't the time complexity of the function that detects the task in the dataset high?

Maybe we need to discuss for user's use case. Thanks for your opinion @harimkang , @kprokofi 
Here is what I wanted for this auto-configuration functionality.

If the user inputs `--train-data-roots`, then the user might want to `automatic auto split` for the dataset.
So, if the user only inputs `--train-data-roots`, I think OTX build must trigger the `auto_config`.

In this case, if the user additionally gives `task` or `model`, then OTX build overwrites the `task`, and `mode` to pre-detected ones.
That's the reason why I use only `--train-data-roots` to trigger the `auto config`.

BTW, there are also some users who want to not use `auto-config` even if they give only `--train-data-roots`.
So, in this circumstance, I think we need to think more about what is our default setting.